src/main/java/com/appdynamics/monitors/azure/AzureMonitor.java

```diff
@@ -8,139 +8,43 @@
-                        configuration.getExecutorService().execute(fabricTask);
-                    }
-                    URL metricDefinitions = Utilities.getUrl(
-                            Globals.azureEndpoint +
-                                    resourceNode.get("id").asText() +
-                                    Globals.azureApiMetricDefinitions +
-                                    "?" + Globals.azureApiVersion +
-                                    "=" + config.get(Globals.azureMonitorApiVersion));
-                    JsonNode metricDefinitionResponse = AzureRestOperation.doGet(AuthenticationResults.azureMonitorAuth,metricDefinitions);
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Get Metric Definitions REST API Request: "
-                                + metricDefinitions.toString());
-                        logger.debug("Get Metric Definitions Response JSON: "
-                                + Utilities.prettifyJson(metricDefinitionResponse));
-                    }
-                    assert metricDefinitionResponse != null;
-                    ArrayNode metricDefinitionElements = (ArrayNode) metricDefinitionResponse.get("value");
-                    for(JsonNode metricDefinitionNode:metricDefinitionElements){
-                        if (metricDefinitionNode.get("isDimensionRequired").asText().equals("true")){
-                            logger.info("Dimensions are currently not supported. Skipping "
-                                    + metricDefinitionNode.get("id").asText());
-                        }
-                        else if (Utilities.checkResourceFilter(metricDefinitionNode,resourceFilter)){
-                            logger.info("Ignoring Metric " +
-                                    metricDefinitionNode.get("name").get("value").asText() +
-                                    " for Resource " + metricDefinitionNode.get("resourceId"));
-                        }
-                        else {
-                            AzureMonitorTask monitorTask = new AzureMonitorTask(
-                                    configuration,
-                                    resourceNode,
-                                    AuthenticationResults.azureMonitorAuth,
-                                    metricDefinitionNode.get("name").get("value").asText());
-                            configuration.getExecutorService().execute(monitorTask);
-                        }
-                    }
-                }
-                    logger.info("Finished gathering Metrics");
-            }
-            else { logger.error("The config.yml is not loaded due to previous errors.The task will not run"); }
+    @Override
+    protected void doRun(TasksExecutionServiceProvider tasksExecutionServiceProvider) {
+        List<Map<String,?>> subscriptions = (List<Map<String,?>>)getContextConfiguration().getConfigYml().get("subscriptions");
```

Comment:
A cleaner approach would be to create a global called configYml, for example, and fetching it from the MonitorContextConfiguration once. I see you've done it several times. You can then use this configYml object to get your subscriptions, serviceFabrics etc.